common/buildcraft/lib/misc/StackUtil.java

```diff
@@ -298,6 +304,12 @@ public static boolean isMatchingItem(@Nonnull final ItemStack base, @Nonnull fin
         if (base.getItem() != comparison.getItem()) {
             return false;
         }
+
+        Set<String> matchingNBT = matchingNBTRegistry.get(base.getItem());
+        if (matchingNBT != null && !isNbtSubsetEquals(base, comparison, matchingNBT)) {
+            return false;
+        }
```

Comment:
Can you move this to an `else` statement below `if (matchNBT)` so that we still perform strict comparison first?